File: eventhub_mobile/lib/models/invitation.dart
class Invitation {
  final int id;
  final int eventId;
  final String eventTitle;
  final String guestName;
  final String guestEmail;
  final String qrCode;
  final String status; // "PENDING" | "USED"

  const Invitation({
    required this.id,
    required this.eventId,
    required this.eventTitle,
    required this.guestName,
    required this.guestEmail,
    required this.qrCode,
    required this.status,
  });

  bool get isPending => status == 'PENDING';

  factory Invitation.fromJson(Map<String, dynamic> json) => Invitation(
        id: json['id'] as int,
        eventId: json['eventId'] as int,
        eventTitle: json['eventTitle'] as String,
        guestName: json['guestName'] as String,
        guestEmail: json['guestEmail'] as String,
        qrCode: json['qrCode'] as String,
        status: json['status'] as String,
      );
}

File: src/main/java/com/eventhub/model/Invitation.java
package com.eventhub.model;

import jakarta.persistence.*;

@Entity
@Table(name = "invitations")
public class Invitation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "guest_id", nullable = false)
    private User guest;

    @Column(nullable = false, unique = true)
    private String qrCode;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    public Invitation() {}

    public Invitation(Long id, Event event, User guest, String qrCode, Status status) {
        this.id = id;
        this.event = event;
        this.guest = guest;
        this.qrCode = qrCode;
        this.status = status;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public User getGuest() {
        return guest;
    }

    public void setGuest(User guest) {
        this.guest = guest;
    }

    public String getQrCode() {
        return qrCode;
    }

    public void setQrCode(String qrCode) {
        this.qrCode = qrCode;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public enum Status {
        PENDING, USED
    }
}
